src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/AssemblyContigAlignmentsConfigPicker.java

```diff
@@ -404,12 +499,54 @@ private static double computeTigExplainQualOfOneConfiguration(final List<Alignme
         return tigExplainedQual;
     }
 
-    //==================================================================================================================
+    /**
+     * For contigs with more than 1 best-scored configurations as determined by
+     * {@link #pickBestConfigurations(AlignedContig, Set, Double)},
+     * save the contigs that has one and only one configuration that
+     * has all mapping quality strictly above the specified {@code threshold}.
+     * Example:
+     *  if a contig has two "optimal" configurations with MQ's {10, 60, 60}, and {60, 60},
+     *  this function will favor/pick the {60, 60} configuration if the {@code threshold} is 10,
+     *  hence breaking the degeneracy;
+     *  on the other hand if the {@code mqThreshold} is passed in as any value below 10, say 0,
+     *  then this function returns all both original {@code differentRepresentationsForOneContig}s
+     */
+    @VisibleForTesting
+    static List<GoodAndBadMappings> filterSecondaryConfigurationsByMappingQualityThreshold(
+            final List<GoodAndBadMappings> differentRepresentationsForOneContig,
+            final int mqThreshold) {
+
+        if ( differentRepresentationsForOneContig.size() == 1) {
+            return differentRepresentationsForOneContig;
+        } else {
+            final List<GoodAndBadMappings> collect = Utils.stream(differentRepresentationsForOneContig)
+                    .filter(rep -> rep.goodMappings.stream().mapToInt(ai -> ai.mapQual).min().orElse(mqThreshold) > mqThreshold)
+                    .collect(Collectors.toList());
+            if ( collect.size()!=1 ) {
+                return differentRepresentationsForOneContig;
+            } else {
+                return collect;
+            }
+        }
+    }
+
+    // step 3: reconstruction from the picked configurations ===========================================================
 
     /**
      * Reconstructs (possibly more than one) {@link AlignedContig} based on
      * the given best-scored configuration(s) in {@code nameSeqAndBestConfigurationsOfOneAssemblyContig}.
      *
+     * Essentially, it
+     * <ul>
+     *     <li>
+     *         splits alignments that contain large gaps into multiple child alignments
+     *     </li>
+     *     <li>
+     *         performs a final round of cleanup by moving alignments that offers low uniqueness to bad mappings
```

Comment:
Can you reword this? I'm not sure what "moving alignments that offers low uniqueness to bad mappings" means